traits/trait_types.py

```diff
@@ -2593,11 +2590,13 @@ class Person(HasTraits):
     #: The default value type to use (i.e. 'constant'):
     default_value_type = DefaultValue.constant
 
-    def __init__(self, *values, **metadata):
-
-        if (len(values) == 1) and (type(values[0]) in SequenceTypes):
-            values = values[0]
-        self.values = values[:]
+    def __init__(self, values, **metadata):
+        if not isinstance(values, SequenceTypes):
+            raise TypeError(
+                "Legal values should be provided via a list, "
```

Comment:
The mention of tuple is excluded here...making the error message inconsistent with the actual check.

Reply:
Maybe what we want to do is mimic what's in `BaseEnum`: do an explicit `isinstance` check to _exclude_ `str`, `bytes` and `bytearray`, but otherwise do no checking and rely on duck typing.

Reply:
Now I am thinking we should just check `isinstance(values, list)`.

Reply:
Race condition: I did not see @mdickinson's message when I posted mine. Duck-typing with the exclusion of strings and friends would do too.

Reply:
So looking at the code, I think we could replace `self.values = values[:]` with `self.values = list(values)` (after doing the check to exclude `str` and friends).

Reply:
Done.